src/openvic-simulation/economy/FactoryProducer.cpp

```diff
@@ -37,3 +44,71 @@ fixed_point_t FactoryProducer::get_average_profitability_last_seven_days() const
 
 	return sum / (1 + profit_history_current);
 }
+
+void FactoryProducer::produce(
+	const fixed_point_t input_modifier, const fixed_point_t throughput_modifier, const fixed_point_t output_modifier
+) {
+	std::map<PopType const* const, fixed_point_t> employees_per_job {};
+	for (const Job job : production_type.get_jobs()) {
+		PopType const* const pop_type = job.get_pop_type();
+		employees_per_job[pop_type] = 0;
+	}
+
+	for (std::pair<Pop const* const, const fixed_point_t> pop_and_number_employed : employees) {
+		Pop const* const pop = pop_and_number_employed.first;
+		const fixed_point_t number_employed = pop_and_number_employed.second;
+		PopType const* const pop_type = &(pop->get_type());
+		employees_per_job[pop_type] += number_employed;
+	}
+
+	fixed_point_t my_input_multiplier = input_modifier, my_throughput_multiplier = throughput_modifier,
+				  my_output_multiplier = output_modifier;
+	for (const Job job : production_type.get_jobs()) {
+		PopType const* const pop_type = job.get_pop_type();
+		const fixed_point_t number_of_employees_in_job = employees_per_job[pop_type];
+		switch (job.get_effect_type()) {
+		case Job::effect_t::INPUT:
+			my_input_multiplier += job.get_effect_multiplier() * number_of_employees_in_job /
+				(size_multiplier * production_type.get_base_workforce_size());
+			break;
+		case Job::effect_t::OUTPUT:
+			my_output_multiplier += job.get_effect_multiplier() * number_of_employees_in_job /
+				(size_multiplier * production_type.get_base_workforce_size());
+			break;
+		case Job::effect_t::THROUGHPUT:
+			my_throughput_multiplier *=
+				job.get_effect_multiplier() * number_of_employees_in_job / production_type.get_base_workforce_size();
```

Comment:
Note size_multiplier is left out here since it would be cancelled by multiplying the base input & output quantities.